ui/button.js

```diff
@@ -30,389 +30,317 @@
 		// Browser globals
 		factory( jQuery );
 	}
-}(function( $ ) {
+}( function( $ ) {
 
-var lastActive,
-	baseClasses = "ui-button ui-widget ui-state-default ui-corner-all",
-	typeClasses = "ui-button-icons-only ui-button-icon-only ui-button-text-icons ui-button-text-icon-primary ui-button-text-icon-secondary ui-button-text-only",
-	formResetHandler = function() {
+var formResetHandler = function() {
 		var form = $( this );
-		setTimeout(function() {
-			form.find( ":ui-button" ).button( "refresh" );
-		}, 1 );
+
+		// Wait for the form reset to actually happen before refreshing
+		setTimeout( function() {
+
+			// We find .ui-button first then filter by :ui-button because doing a
+			// widget pseudo selectors are very very slow but we need to filter out
+			// css only buttons
```

Comment:
Fixed multiple issues:

```js
// We find .ui-button first, then filter by :ui-button,
// because widget pseudo selectors are very very slow, but
// we need to filter out CSS-only buttons
```